### ISSUE TITLE
`Move left/right` commands are sometimes undoable with their `Move right/left` counterpart

### ISSUE THREAD

**Issue by reuben** [Gregoor/soy#6]
**STR:**

Open a file with e.g. the following contents:

```js
// orig
const foo = [
  ["first", 23],
  ["second", ≤52≥]
];
```
(`≤` indicates selection start, `≥` selection end)

Move the cursor to `52`, then use `Move left`:

```js
// step1
const foo = [
  ["first", 23],
  [≤52≥, "second"]
];
```

`Move left` again:

```js
// step2
const foo = [
  ≤[52, "second"]≥,
  ["first", 23]
];
```

At this point, there is no way to return to the `orig` state using `Move right`, you have to use `Undo`. The way I conceptualize it is that in `step2` there's an implicit `Expand selection` happening which I wasn't expecting. As I write this I'm not quite sure if this is an artificial case but I'm filing since it caught me by surprise at first.

**Comment by Gregoor** [comment]
You are correct that it expands until it finds something moveable. A principle I took from Tofu is that I prefer it doing something over nothing, since something can be undone whereas nothing is.. nothing. But true, it is surprising at first (doubly so since it's not documented).

So I guess the question is whether one gets used to it or if the surprise will stay? In which case I'd wanna get rid of it.

**Comment by reuben** [comment]
For me the core surprise is related to not always being able to undo a `left` command by using a`right` command, or undo an `up` command with a `down` command. This grammar is a bit burned into my head. But yes, I think I'll get used to this. And if not, I'll bug you here some more 🥳

**Comment by Gregoor** [comment]
I feel you, it also bugs me that it breaks symmetry.

I've made many decisions in Tofu that lean in favor of ~some powerful action, over nothing or the merely-literal. That fucks with predictability until one has a good mental model of the editor (at that point I'd hope one would not want to go back, but that's a very VIM way to think I guess). Maybe Soy should lean more towards predictability. Food for thought!

**Comment by Gregoor** [comment]
Idea from Reuben to retain symmetry: remember when a move needed to extend the selection and then use that info to symmetrical undo moves
